dateutil/parser/_parser.py

```diff
@@ -49,7 +49,17 @@
 from .. import relativedelta
 from .. import tz
 
-__all__ = ["parse", "parserinfo"]
+__all__ = ["parse", "parserinfo", "ParserError"]
+
+
+class ParserError(ValueError):
+    def __str__(self):
+        try:
+            return self.args[0] % self.args[1:]
+        except (TypeError, IndexError):
+            return super(ParserError, self).__str__()
+
+    __repr__ = __str__
```

Comment:
This is not quite appropriate, because `str` returns only the message portion of the error and does not include the class:

```python
>>> str(ValueError("Test", "test"))
"('Test', 'test')"
>>> repr(ValueError("Test", "test"))
"ValueError('Test', 'test')"
```

So this should be maybe:

```python
def __repr__(self):
    return "%s(%s)" % (self.__class__.__name__, str(self))
```